src/librbd/cache/SharedPersistentObjectCacherFile.cc

```diff
@@ -52,7 +52,15 @@ void SyncFile::open(Context *on_finish) {
   on_finish->complete(0);
 }
 
-void SyncFile::open() {
+int SyncFile::open_file() {
+  m_fd = ::open(m_name.c_str(), O_RDONLY);
+  if(m_fd == -1) {
+    lderr(cct) << "open fails : " << std::strerror(errno) << dendl;
+  }
+  return m_fd;
+}
+
+void SyncFile::create() {
```

Comment:
at my testing environments, i have seen one case : when fio closed,  it will hang. The reason is that current read operation stunk into dead loop once open() operation fails. 
why don't directly return error code.